### ISSUE TITLE
Not working on firefox and explorer

### ISSUE THREAD

**Issue by gokaygok** [credativUK/redmine_image_clipboard_paste#7]
it worsk on chrome but not works on firefox and IE

Environment:
  Redmine version                2.5.1.stable
  Ruby version                   1.9.3-p545 (2014-02-24) [x86_64-linux]
  Rails version                  3.2.17
  Environment                    production
  Database adapter               Mysql2
SCM:
  Subversion                     1.6.12
  Mercurial                      1.9.1
  Git                            1.7.5.4
  Filesystem  
Redmine plugins:
  redmine_backlogs               v1.0.6
  redmine_dmsf                   1.4.7 stable
  redmine_extended_spent_time    0.0.3
  redmine_image_clipboard_paste  1.0.0
  redmine_spent_time             2.6.5

**Comment by DJUserkent** [comment]
For me - works on Chrome 35 fine, not on Firefox 30:    (I'm on Ubuntu).
FF fails in all of 'New issue', 'add to existing issue'.  'add to Wiki'

**Comment by tornoz** [comment]
Firefox (and IE) does not manage the clipboard in the same way, so it is not possible to do everything Webkit-based browsers can do for the moment.

I have pushed a fix that allows copy-pasting images in Firefox (and possibly IE, I have not tested) from web-browsers. It is not possible, however, to copy-paste from a local file, as Firefox only allows to get a link to the copied image, not the data like Chrome does.

**Comment by DJUserkent** [comment]
Tomoz, to be super sure we understand your comment:

> I have pushed a fix that allows copy-pasting images in Firefox ...
>  I assume this means ' copy-pasting from clipboard of an image saved there'.
> 
> It is not possible, however, to copy-paste from a local file ...

Which is not about using the clipboard.

Shout if this understanding is wrong  :<)

**Comment by tornoz** [comment]
-  Chrome allows to copy-paste an image from your computer as it retrieve the raw data from the clipboard. This is not possible in Firefox yet.
-  Copy-pasting an image from the browser, however, is possible in Firefox if the original file can be accessed directly by an http request.

If there are some parts that remains unclear, please ask.

**Comment by DJUserkent** [comment]
>  is possible in Firefox if the original file can be accessed 

what is the 'original file' here?

Because the whole point of this plugin, is that you don't have to start with a file of an image, you start with just a clipboard image from e.g. a  screenshot.

Maybe you mean:
"Firefox does allow a less useful alternative:  to take an image file that already exists (ie not from your clipboard)  , and to paste that into Redmine - but only if the original file can be directly accessed by an HTTP request"

**Comment by tornoz** [comment]
Yep, that's it. Firefox doesn't allow access to raw data in the clipboard for security reasons.
I understand that this is less interesting but that's all we can do for the moment.

**Comment by 646b** [comment]
https://github.com/credativUK/redmine_image_clipboard_paste/pull/23
